src/dispdrivers/vgacontroller.cpp

```diff
@@ -203,6 +265,39 @@ void IRAM_ATTR VGAController::rawFillRow(int y, int x1, int x2, uint8_t pattern)
 }
 
 
+// parameters not checked
+void IRAM_ATTR VGAController::rawORRow(int y, int x1, int x2, uint8_t pattern)
+{
+  auto row = m_viewPort[y];
+  // naive implementation - just do whole row iteratively
+  for (int x = x1; x <= x2; ++x) {
+    VGA_PIXELINROW(row, x) |= pattern;
+  }
+}
```

Comment:
these three new functions allow the various new paint modes to work for "fill" operations

they're all currently naive implementations, setting a pixel at a time.  they could be optimised to operate on 32-bit words in a similar-ish manner to how `rawFillRow` works (by doing pixel operations for un-aligned bytes, and word operations for all the aligned bytes)